wgpu-core/src/device/resource.rs

```diff
@@ -235,11 +238,13 @@ impl Device {
         let alignments = adapter.raw.capabilities.alignments.clone();
         let downlevel = adapter.raw.capabilities.downlevel.clone();
 
-        #[cfg(feature = "indirect-validation")]
-        let indirect_validation = if downlevel
-            .flags
-            .contains(wgt::DownlevelFlags::INDIRECT_EXECUTION)
-        {
+        let enable_indirect_validation = instance_flags
+            .contains(wgt::InstanceFlags::VALIDATION_INDIRECT_CALL)
+            && downlevel
+                .flags
+                .contains(wgt::DownlevelFlags::INDIRECT_EXECUTION);
```

Comment:
Thought: can there be a device that has indirect execution, but not compute shaders?

Reply:
Only our GL backend sets those 2 features differently (and metal, but it always supports compute shaders):

https://github.com/gfx-rs/wgpu/blob/a734e86041beb383e8d330b5230e9a05f2bcd288/wgpu-hal/src/gles/adapter.rs#L312-L313

https://github.com/gfx-rs/wgpu/blob/a734e86041beb383e8d330b5230e9a05f2bcd288/wgpu-hal/src/gles/adapter.rs#L380-L381

The thing is, I don't think it's right that we check for `GL_ARB_multi_draw_indirect` as it doesn't provide indirect dispatches.

Reply:
I will remove `GL_ARB_multi_draw_indirect` and document that `INDIRECT_EXECUTION` implies presence of `COMPUTE_SHADERS`.

Reply:
Actually it seems `GL_ARB_compute_shader` enables indirect dispatches.